src/Fantomas/FakeHelpers.fs

```diff
@@ -55,75 +53,94 @@ let formatFileAsync config (file : string) =
                 if not isValid  then
                     raise <| FormatException "Formatted content is not valid F# code"
 
-                let tempFile = Path.GetTempFileName()
-                use output = new StreamWriter(tempFile)
-                output.Write formattedContent
-                output.Flush()
-
-                return Formatted(file, tempFile)
+                return Formatted(filename=file, formattedContent=formattedContent)
             else
-                return Unchanged file
+                return Unchanged(filename=file)
         with
         | ex -> return Error(file, ex)
     }
 
+let formatFileAsync config (file : string) =
+    let originalContent = File.ReadAllText file
+    async {
+        let! formated = originalContent |> formatContentAsync config file
+        return formated
+    }
+
 let formatFilesAsync config files =
     files
     |> Seq.map (formatFileAsync config)
     |> Async.Parallel
 
-let internal removeTemporaryFiles formatResult =
-    match formatResult with
-    | Formatted(_, tmp) -> File.Delete(tmp)
-    | _ -> ()
-
 let formatCode config files =
     async {
         let! results = files |> formatFilesAsync config
         
-        try
-            // Check for formatting errors:
-            let errors =
-                results
-                |> Array.choose (fun x ->
-                    match x with
-                    | Error(file, ex) -> Some(file, Some(ex))
-                    | _ -> None)
-
-            if not <| Array.isEmpty errors then
-                raise <| CodeFormatException errors
-
-            // Overwritte source files with formatted content
-            let result =
-                results
-                |> Array.choose (fun x ->
-                    match x with
-                    | Formatted(source, tmp) ->
-                        File.Copy(tmp, source, true)
-                        Some source
-                    | _ -> None)
-                
-            return result
-        finally
-            results |> Array.iter removeTemporaryFiles
-    }
+        // Check for formatting errors:
+        let errors =
+            results
+            |> Array.choose (fun x ->
+                match x with
+                | Error(file, ex) -> Some(file, Some(ex))
+                | _ -> None)
+
+        if not <| Array.isEmpty errors then
+            raise <| CodeFormatException errors
+
+        // Overwritte source files with formatted content
+        let result =
+            results
+            |> Array.choose (fun x ->
+                match x with
+                | Formatted(source, formatted) ->
+                    File.WriteAllText(source, formatted)
+                    Some source
+                | _ -> None)
 
+        return result
+    }
 
-let checkCode config files =
+type CheckResult =
```

Comment:
Haha! Nice one, it's like you read my mind with this type. This way it's easier to do reporting on errors/changes 😄